### PR TITLE
Modify removeHs so that it does not remove degree 0 atoms

### LINKED ISSUE #2086: Crash in RemoveHs
Here's the reproducible (unfortunately not simple):
```
In [1]: from rdkit import Chem

In [2]: m2 = Chem.MolFromSmiles('C1CCCC1')

In [3]: mh2 = Chem.RWMol(Chem.AddHs(m2))

In [4]: mh2.RemoveBond(4,13)

In [5]: mh2.GetAtomWithIdx(4).SetNoImplicit(True)

In [6]: mh2.GetAtomWithIdx(13).SetNoImplicit(True)

In [7]: Chem.SanitizeMol(mh2)
Out[7]: rdkit.Chem.rdmolops.SanitizeFlags.SANITIZE_NONE

In [8]: mh2.Debug()
Atoms:
        0 6 C chg: 0  deg: 4 exp: 4 imp: 0 hyb: 4 arom?: 0 chi: 0
        1 6 C chg: 0  deg: 4 exp: 4 imp: 0 hyb: 4 arom?: 0 chi: 0
        2 6 C chg: 0  deg: 4 exp: 4 imp: 0 hyb: 4 arom?: 0 chi: 0
        3 6 C chg: 0  deg: 4 exp: 4 imp: 0 hyb: 4 arom?: 0 chi: 0
        4 6 C chg: 0  deg: 3 exp: 3 imp: 0 hyb: 4 arom?: 0 chi: 0 rad: 1
        5 1 H chg: 0  deg: 1 exp: 1 imp: 0 hyb: 1 arom?: 0 chi: 0
        6 1 H chg: 0  deg: 1 exp: 1 imp: 0 hyb: 1 arom?: 0 chi: 0
        7 1 H chg: 0  deg: 1 exp: 1 imp: 0 hyb: 1 arom?: 0 chi: 0
        8 1 H chg: 0  deg: 1 exp: 1 imp: 0 hyb: 1 arom?: 0 chi: 0
        9 1 H chg: 0  deg: 1 exp: 1 imp: 0 hyb: 1 arom?: 0 chi: 0
        10 1 H chg: 0  deg: 1 exp: 1 imp: 0 hyb: 1 arom?: 0 chi: 0
        11 1 H chg: 0  deg: 1 exp: 1 imp: 0 hyb: 1 arom?: 0 chi: 0
        12 1 H chg: 0  deg: 1 exp: 1 imp: 0 hyb: 1 arom?: 0 chi: 0
        13 1 H chg: 0  deg: 0 exp: 0 imp: 0 hyb: 1 arom?: 0 chi: 0 rad: 1
        14 1 H chg: 0  deg: 1 exp: 1 imp: 0 hyb: 1 arom?: 0 chi: 0
Bonds:
        0 0->1 order: 1 conj?: 0 aromatic?: 0
        1 1->2 order: 1 conj?: 0 aromatic?: 0
        2 2->3 order: 1 conj?: 0 aromatic?: 0
        3 3->4 order: 1 conj?: 0 aromatic?: 0
        4 4->0 order: 1 conj?: 0 aromatic?: 0
        5 0->5 order: 1 conj?: 0 aromatic?: 0
        6 0->6 order: 1 conj?: 0 aromatic?: 0
        7 1->7 order: 1 conj?: 0 aromatic?: 0
        8 1->8 order: 1 conj?: 0 aromatic?: 0
        9 2->9 order: 1 conj?: 0 aromatic?: 0
        10 2->10 order: 1 conj?: 0 aromatic?: 0
        11 3->11 order: 1 conj?: 0 aromatic?: 0
        12 3->12 order: 1 conj?: 0 aromatic?: 0
        13 4->14 order: 1 conj?: 0 aromatic?: 0

In [9]: Chem.RemoveHs(mh2)
(base)
glandrum@GregL-KNIME MINGW64 ~/RDKit_git/builds/win64_modern (feat/github1850)

```

### PR DESCRIPTION
**Pull request by greglandrum**
Update the docs at the same time.

Fixes #2086

### CODE REVIEW

File: Code/GraphMol/AddHs.cpp [hunk 1]
```diff
@@ -562,7 +561,7 @@ void removeHs(RWMol &mol, bool implicitOnly, bool updateExplicitCount,
     Atom *atom = mol.getAtomWithIdx(currIdx);
     idxMap[origIdx] = currIdx;
     ++origIdx;
-    if (atom->getAtomicNum() == 1) {
+    if (atom->getAtomicNum() == 1 && atom->getDegree()) {
```
Comment (bp-kelley) [inline Code/GraphMol/AddHs.cpp]: I think this is a time for a warning.  This is mildly unexpected outcome that removeHs doesn't ;)

I.e. WARNING: not removing zero degree hydrogen and index 10

This is now really removeConnectedHs, correct?

Comment (bp-kelley) [reply Code/GraphMol/AddHs.cpp]: What happens with [H][H] btw, does only one get removed?

Comment (greglandrum) [reply Code/GraphMol/AddHs.cpp]: good point about the warning.
I'll add a test case for `[H][H]`

Comment (greglandrum) [reply Code/GraphMol/AddHs.cpp]: I should be more careful when answering things from my phone... :-)
The `[H][H]` case is already documented: https://github.com/rdkit/rdkit/blob/master/Code/GraphMol/MolOps.h#L209
and tested: https://github.com/rdkit/rdkit/blob/master/Code/GraphMol/molopstest.cpp#L613

**Pull request merged**